### PR TITLE
Added a search bar in the navigation as required

### LINKED ISSUE #202: Add search feature
When having a large number of OTP sites, searching manually by Icon is annoying (especially when having multiple applications with the same Icon).

A quick ability to search (magnifying glass in the menu bar) would really help out here. As you type letters, the list of icons shown should change to only show the list that match what you entered.

### PR DESCRIPTION
**Pull request by mul1sh**
@justin-stephenson  This fixes #202  with UI tests and a bug fix for the UI tests not added to the build settings

### CODE REVIEW

**Comment by mul1sh** [comment]
@justin-stephenson this is ready for review 🙂

**Comment by justin-stephenson** [comment]
Thank you @mul1sh this looks like a useful feature, I hope to have time to test soon and review this PR.

**Comment by mul1sh** [comment]
@justin-stephenson no worries take your time 🙂, you can also check out a screenshot of how it works here https://github.com/freeotp/freeotp-ios/issues/202#issuecomment-730267275

Many thanks

**Comment by mul1sh** [comment]
@justin-stephenson  I've updated my PR to match the contributions guidelines. And so the search UI changes are as shown in the screenshot below

appVideo

Thanks 🙂

**Comment by justin-stephenson** [comment]
Apologies for the delayed review @mul1sh, the search functionality works great except one issue with attempting to remove a token(swipe right) from the filtered tokens list.

It looks like the `handleSwipe` method may need to be modified to address this issue. I see the following error in my testing:

**2020-12-09 15:14:48.880442-0500 FreeOTP[511:39527] [UICollectionView] Invalid update: invalid number of items in section 0.  The number of items contained in an existing section after the update (2) must be equal to the number of items contained in that section before the update (2), plus or minus the number of items inserted or deleted from that section (0 inserted, 1 deleted) and plus or minus the number of items moved into or out of that section (0 moved in, 0 moved out). - will perform reloadData. UICollectionView instance: <UICollectionView: 0x106018200; frame = (0 0; 375 591); clipsToBounds = YES; opaque = NO; autoresize = W+H; gestureRecognizers = <NSArray: 0x2828190b0>; layer = <CALayer: 0x28262dba0>; contentOffset: {0, 0}; contentSize: {375, 248.46153846153845}; adjustedContentInset: {0, 0, 216, 0}; layout: <UICollectionViewFlowLayout: 0x10581e6e0>; dataSource: <FreeOTP.TokensViewController: 0x104f0abf0>>; currentUpdate: [UICollectionViewUpdate - 0x108812ac0: old:<UICollectionViewData: 0x281f7c380> new<UICollectionViewData: 0x281f60d20> items:<(
    "D(0,0)"
)>]**

Thank you again for the PR!

File: FreeOTP/TokenStore.swift [hunk 1]
```diff
@@ -143,6 +148,10 @@ open class TokenStore : NSObject {
 
         return nil
     }
+    
+   
+
+
```
Comment (justin-stephenson) [inline FreeOTP/TokenStore.swift:154]: Please remove this

**Comment by justin-stephenson** [comment]
Nice UITest by the way, that is quite useful :)

**Comment by justin-stephenson** [comment]
@mul1sh It would also be great if the UITest can run on the simulator, adding some test tokens into the App before executing the test. I can help with this if needed.

**Comment by mul1sh** [comment]
@justin-stephenson thank you for the awesome feedback, I appreciate it 😃.  I've fixed the bug you got when trying to remove a token.

Also yes I would definitely like to know how to add test tokens before running the UI test, so your help would be highly appreciated.

**Comment by justin-stephenson** [comment]
> @justin-stephenson thank you for the awesome feedback, I appreciate it 😃. I've fixed the bug you got when trying to remove a token.
> 
> Also yes I would definitely like to know how to add test tokens before running the UI test, so your help would be highly appreciated.

Thanks, this looks good now. Merging.

**Comment by mul1sh** [comment]
Awesome thanks @justin-stephenson glad you like it 😃

**Comment by mul1sh** [comment]
@justin-stephenson Thanks for alerting me, i'll submit a PR for this later today 🙂

**Pull request merged**